### ISSUE TITLE
[NETSDKE2E][Workload Sets] With manifest mode and workload installed, Unhandled exception occurs when running `dotnet workload --info` using global.json 

### ISSUE THREAD

**Issue by NicoleWang001** [dotnet/sdk#42581]
### Describe the bug
With manifest mode and workload installed, Unhandled exception occurs when running '`dotnet workload --info` using global.json

### To Reproduce
1. Install .NET 8.0.400 SDK
2. Install workload eg. aspire 
`    dotnet workload install aspire`
![image](https://github.com/user-attachments/assets/0b5dd3fe-5657-47c4-8bc6-68839412a930)

3.  New a global.json and add workloadverion to this file under a new folder
```
    c:\tt\TestWithGloabalJson
    cd c:\tt\TestWithGloabalJson
    dotnet new global.json
```
```
    {
      "sdk": {
        "version": "8.0.400",
        "workloadVersion": "8.0.400"
      }
    }
```
4. Run `dotnet workload --info` and observe

### Expected Result
Proper suggestion without any exception. eg. '`dotnet workload list`'
![image](https://github.com/user-attachments/assets/1f8ef6ed-9131-4dd2-a03b-c49f9e0ebaee)

### Actual Result
Unhandled exception:
![image](https://github.com/user-attachments/assets/1c2e000e-57a5-4c68-ad99-bee0666b2655)

**Comment by Forgind** [comment]
This issue report looks correct to me, but I consider it slightly lower priority because although it shouldn't be an unhandled exception with a call stack, it is still an error scenario, and the error message is correct.

**Comment by vdanche** [comment]
@Forgind Currently this bug not fix in net8.0.402. right?
![image](https://github.com/user-attachments/assets/8a5f1f21-3b5d-4068-a811-85433a46b786)

**Comment by Forgind** [comment]
> @Forgind Currently this bug not fix in net8.0.402. right? ![image](https://private-user-images.githubusercontent.com/65637796/360804340-8a5f1f21-3b5d-4068-a811-85433a46b786.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQ0MjI1NjQsIm5iZiI6MTcyNDQyMjI2NCwicGF0aCI6Ii82NTYzNzc5Ni8zNjA4MDQzNDAtOGE1ZjFmMjEtM2I1ZC00MDY4LWE4MTEtODU0MzNhNDZiNzg2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODIzVDE0MTEwNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQwNGY4NmZjYTRkOTBlOTM4NDg4NWE0MmQ3ZTMwZjAzMDdhYzk1MzhiOTU0OTQyNDk5ODBiZDMyYTA1Mjc5Y2ImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.FyCI1_3CHTU1dR13FwZWXdC1vfMj6RPH9MhICMLauNE)

It was merged into main last week, so it shouldn't be fixed until 9.0

**Comment by vdanche** [comment]
@Forgind the latest message for NET9 RC2 as below when running dotnet workload --info using globaljson:

dotnet workload --info, error "Unhandled exception: System.IO.FileNotFoundException..." still remains，right？

![image](https://github.com/user-attachments/assets/f3317f76-096f-417e-9f5f-517733e7b4a3)

dotnet --info， "Workload version 9.0.100, which was specified in..." not display in dotnet --info, is it expected?
![image](https://github.com/user-attachments/assets/19a5bca7-7104-41c7-b903-d3f2156f54ca)

dotnet workload --version
![image](https://github.com/user-attachments/assets/b37f08f4-00d9-4bad-b45f-2cf5dd1336eb)

note:
dotnet workload list
![image](https://github.com/user-attachments/assets/72a15296-9658-4dd3-8a07-919bd2734420)

**Comment by Forgind** [comment]
> @Forgind the latest message for NET9 RC2 as below when running dotnet workload --info using globaljson:
> 
> dotnet workload --info, error "Unhandled exception: System.IO.FileNotFoundException..." still remains，right？
> 
> ![image](https://private-user-images.githubusercontent.com/65637796/368515250-f3317f76-096f-417e-9f5f-517733e7b4a3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjY2NzkyOTEsIm5iZiI6MTcyNjY3ODk5MSwicGF0aCI6Ii82NTYzNzc5Ni8zNjg1MTUyNTAtZjMzMTdmNzYtMDk2Zi00MTdlLTlmNWYtNTE3NzMzZTdiNGEzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTE4VDE3MDMxMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTBkNGYxYWYyMDU3NTdlOGMzNDhhOTI0MTcyYjA4MmE0MGU5MTg3NzQxYTM5N2NmNTdkYTQ0NWU4NDA3NGM4M2YmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.azFGqJGvIhqlW67sYl6fYYT7f7sX-FmuNpIm02eS5dM)
> 
> dotnet --info， "Workload version 9.0.100, which was specified in..." not display in dotnet --info, is it expected? ![image](https://private-user-images.githubusercontent.com/65637796/368515411-19a5bca7-7104-41c7-b903-d3f2156f54ca.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjY2NzkyOTEsIm5iZiI6MTcyNjY3ODk5MSwicGF0aCI6Ii82NTYzNzc5Ni8zNjg1MTU0MTEtMTlhNWJjYTctNzEwNC00MWM3LWI5MDMtZDNmMjE1NmY1NGNhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTE4VDE3MDMxMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJlMDAwZjE1ZTg5ZDY4M2Y2Njg2YmM0ZDI1YjJiZGQ5YTBmYWYzMjZjNDdkNDM0MWU2YzNkODQxMTZiNjM1NzkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.ri_IZq3lIEhIYD0Jdd0nmjuSzsCoKYQ0DerBOzEsAlQ)
> 
> dotnet workload --version ![image](https://private-user-images.githubusercontent.com/65637796/368515659-b37f08f4-00d9-4bad-b45f-2cf5dd1336eb.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjY2NzkyOTEsIm5iZiI6MTcyNjY3ODk5MSwicGF0aCI6Ii82NTYzNzc5Ni8zNjg1MTU2NTktYjM3ZjA4ZjQtMDBkOS00YmFkLWI0NWYtMmNmNWRkMTMzNmViLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTE4VDE3MDMxMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWU3NzZmMDkxYmJiMjVlMzNmNjRhMGY1NWViYTJkZGQ2MDg5MzM3OGFkMzE0ZDc0ZmIwZDM4MjNhZGQ4YzBiNzEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.w1x8xXKE4_Q-WpnJZ8rysCPG6gzCYECR90lczLGZLoQ)
> 
> note: dotnet workload list ![image](https://private-user-images.githubusercontent.com/65637796/368517220-72a15296-9658-4dd3-8a07-919bd2734420.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjY2NzkyOTEsIm5iZiI6MTcyNjY3ODk5MSwicGF0aCI6Ii82NTYzNzc5Ni8zNjg1MTcyMjAtNzJhMTUyOTYtOTY1OC00ZGQzLThhMDctOTE5YmQyNzM0NDIwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTE4VDE3MDMxMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQ3YmIyNDcwMWUyYzQzMzFiNTc3NWQyODZjOGU0NTYwN2E0ZjAyNTIxNTk4NTQ0ODAyNDI3ZmZjYTNmZjczYTgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.A1crGsm2LweKbGBd2Z42D4ySCo6VX3R5jKeHqruISsQ)

I thought it should be in 9.0, but apparently my pr made it in after we branched, so it's in main but not release/9.0.1xx

I'd be in favor of backporting it. @marcpopMSFT, do we still have enough time?

**Comment by marcpopMSFT** [comment]
On the one hand, it's a very niche scenario. On the other, we want people to try out workloads in 9.0.100 more broadly. @baronfel @dsplaisted it probably makes sense to take this for GA as the fix looks reasonable enough. Thoughts?
https://github.com/dotnet/sdk/pull/42630/files

**Comment by dsplaisted** [comment]
Here's a PR backporting this to 9.0.1xx: https://github.com/dotnet/sdk/pull/43587

**Comment by Forgind** [comment]
9.0.2xx now, it seems

**Comment by vdanche** [comment]
> 9.0.2xx now, it seems

On net9 rc2 candidate build --9.0.100-rc.2.24474.11: still display as below running dotnet workload --info
![image](https://github.com/user-attachments/assets/35af2b8b-35c0-46a9-8483-4feed26fbdc2)

**Comment by Forgind** [comment]
> > 9.0.2xx now, it seems
> 
> On net9 rc2 candidate build --9.0.100-rc.2.24474.11: still display as below after running dotnet workload --info using globaljson ![image](https://private-user-images.githubusercontent.com/65637796/371486273-35af2b8b-35c0-46a9-8483-4feed26fbdc2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjc0NDE0MjAsIm5iZiI6MTcyNzQ0MTEyMCwicGF0aCI6Ii82NTYzNzc5Ni8zNzE0ODYyNzMtMzVhZjJiOGItMzVjMC00NmE5LTg0ODMtNGZlZWQyNmZiZGMyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTI3VDEyNDUyMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTFkMjE4ZTNhYTJiZDRiMzNhMGI3NWI2ZmJkOGNiYmZlODUyZmYxZGE4MmEwMThlOTIwOGYwN2I0YTg3NTA3NWQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.2KMeVvA17CZNQSz2REc3yzuKfOoZhDuncAhnhLVEGPo)

That's still 9.0.1xx. It won't be fixed until 9.0.2xx assuming the current PR goes through.

**Comment by vdanche** [comment]
Fixed on 9.0.200-preview.0.24512.3 sdk
dotnet workload --info
![Image](https://github.com/user-attachments/assets/bdd1286b-53d8-4272-89ee-01e20d4902cf)
dotnet --info:
![Image](https://github.com/user-attachments/assets/b456ffe3-ff8b-40a5-861d-1a7cb6829344)
